firrtl/src/test/scala/firrtlTests/SerializerSpec.scala

```diff
@@ -176,4 +220,25 @@ class SerializerSpec extends AnyFlatSpec with Matchers {
     SMemTestCircuit.circuit(ReadUnderWrite.New).serialize should include("new")
     SMemTestCircuit.circuit(ReadUnderWrite.Old).serialize should include("old")
   }
+
+  it should "support emitting Probe/RWProbe types and related expressions/statements" in {
+    val probeInt = DefWire(NoInfo, "foo", ProbeType(UIntType(IntWidth(3))))
+    Serializer.serialize(probeInt) should be("wire foo : Probe<UInt<3>>")
+
+    val rwProbeBundle = DefWire(
+      NoInfo,
+      "foo",
+      RWProbeType(
+        BundleType(
+          Seq(
+            Field("bar", Default, UIntType(IntWidth(32)))
+          )
+        )
+      )
+    )
+    Serializer.serialize(rwProbeBundle) should be("wire foo : RWProbe<{ bar : UInt<32>}>")
```

Comment:
I'm not sure where/if this should be considered (how much FIRRTL type-checking should be done in Chisel vs letting MFC check), but FWIW: references can only exist in/as ports, they can't be used in hardware declarations like wire.

Reply:
Good to know. @jackkoenig do you know if is there a good way to check in Chisel that `Probe`/`RWProbe` are not in actual hardware declarations? Or do you have thoughts on if the check should go into firtool?

Reply:
Ports are hardware declarations, but yes we should check this, you can just use the binding to make sure it has a `PortBinding`

Reply:
> you can just use the binding

Can you explain what binding I'm supposed to be looking at? The case class `RW/Probe(underlying: Type)` doesn't seem to have any binding information.

Reply:
(Sounds like something to let firtool reject if someone tries to emit it :+1: and regardless can be added later as needed/motivated :+1:)

Reply:
👍 On this. We should _at least_ not use this in a `Wire` in the example.

You can also consider just serializing a `RWProbeType` and not bothering with it being a `Wire`.